raft/server.cc

```diff
@@ -449,9 +450,13 @@ future<> server_impl::wait_for_entry(entry_id eid, wait_type type, seastar::abor
 }
 
 future<entry_id> server_impl::add_entry_on_leader(command cmd, seastar::abort_source* as) {
+    const auto memory_usage = memory_usage_of(cmd);
+    if (const auto limit = _config.max_log_size_in_bytes - _config.max_snapshot_trailing_bytes; memory_usage > limit) {
+        throw std::runtime_error(fmt::format("command memory usage {} is greater than limit {}", memory_usage, limit));
```

Comment:
I'l merge this check with `max_command_size` (previous PR)

Reply:
maybe we should measure `max_log_size_in_bytes` and `_config.max_snapshot_trailing_bytes` independently, so that the final limit would be `max_log_size_in_bytes + _config.max_snapshot_trailing_bytes`, error messages would be clearer.

Reply:
added more clear error message